File: functional-tests/functional/python/test/strings_test.py
# Copyright (C) 2016-2025 HERE Europe B.V.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0
# License-Filename: LICENSE

"""String type mapping tests for the Python (pybind11) bindings."""

import functional
from test.StaticStringMethods import StaticStringMethods
from test.StringsWithCstring import StringsWithCstring
from test.CppRefReturnType import CppRefReturnType

import pytest


class TestStaticStringMethods:
    def test_return_input_string(self):
        assert StaticStringMethods.return_input_string("abc") == "abc"

    def test_concatenate_strings(self):
        assert StaticStringMethods.concatenate_strings("a", "b") == "ab"

    def test_return_hello_string(self):
        assert StaticStringMethods.return_hello_string() == "hello"

    def test_return_empty(self):
        assert StaticStringMethods.return_empty() == ""


class TestStringsWithCstring:
    def test_return_input_string_type(self):
        assert StringsWithCstring.return_input_string_type("x") == "x"

    def test_return_input_string(self):
        assert StringsWithCstring.return_input_string("y") == "y"


class TestCppRefReturnType:
    def test_string_ref(self):
        assert CppRefReturnType.string_ref() == "nonsense"

    def test_string_property(self):
        assert CppRefReturnType.string_property() == "nonsense"

File: functional-tests/functional/js/tests/strings.test.mjs
import assert from "node:assert/strict";
import test from "node:test";
import { loadPackage } from "./load-package.mjs";

const module = await loadPackage("test");

test("static string methods round-trip values", () => {
  const methods = module.StaticStringMethods;

  assert.equal(methods.returnInputString("abc"), "abc");
  assert.equal(methods.concatenateStrings("a", "b"), "ab");
  assert.equal(methods.returnHelloString(), "hello");
  assert.equal(methods.returnEmpty(), "");
});

test("C string parameters and returns are converted", () => {
  const methods = module.StringsWithCstring;

  assert.equal(methods.returnInputStringType("x"), "x");
  assert.equal(methods.returnInputString("y"), "y");
});

test("reference-returned strings are readable", () => {
  const type = module.CppRefReturnType;

  assert.equal(type.stringRef(), "nonsense");
  assert.equal(type.stringProperty(), "nonsense");
});